cmake/cmake-utilities.cmake

```diff
@@ -40,6 +40,8 @@ endfunction()
 function(make_utility name srcs)
     make_executable(${name} "${srcs}" "${CSOUNDLIB}")
     add_dependencies(${name} ${CSOUNDLIB})
+    # we use these dependencies in our public headers so link them too
+    target_link_libraries(${name} PUBLIC ${libcsound_PUBLIC_LIBS})
```

Comment:
This doesn't seem like it should be necessary. If we're linking in the libcsound target, it should bring in any dependencies with it?

Reply:
I agree, linking to libcsound would be the sane option. But we aren't allowed to do this I guess, https://github.com/csound/csound/pull/1748#issuecomment-1740207758, so instead, we have to link all dependents to our public dependencies too

Reply:
The comment is correct. The opcodes themselves only need the headers to build against. They get loaded at runtime and just use the Csound API. These utilities don't have the same restriction though.

I think the solution here is that we should have a libcsound interface target that only contains the headers i.e. it's a target that produces no binary. We can use that to link against the opcodes. I'm wondering though, what libcsoudn public libs should be passed to dependent targets? That doesn't seem right to me.

Reply:
Ok, well, I linked the utilities to csound. Still not sure what to do about plugins.

Reply:
I think the right solution would be to build an "internal" `libcsound-internals` that has all headers available, and use that to link plugins and utilities that need internals. Then we can have a "public" `libcsound` that links to `libcsond-internals` but only has public headers available.